presto-main/src/test/java/com/facebook/presto/operator/scalar/TestScalarImplementationValidation.java

```diff
@@ -0,0 +1,89 @@
+package com.facebook.presto.operator.scalar;
+
+import com.facebook.presto.spi.ConnectorSession;
+import com.google.common.collect.ImmutableList;
+import org.testng.annotations.Test;
+
+import java.lang.invoke.MethodHandle;
+import java.util.Optional;
+
+import static com.facebook.presto.util.Reflection.methodHandle;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.fail;
+
+public class TestScalarImplementationValidation
+{
+    private static final MethodHandle STATE_FACTORY = methodHandle(TestScalarImplementationValidation.class, "createState");
+
+    @Test
+    public void testConnectorSessionPosition()
+    {
+        // Without cached instance factory
+        ScalarFunctionImplementation validFunction = new ScalarFunctionImplementation(
```

Comment:
This `ScalarFunctionImplementation` seems like it's not used, but based on offline discussion @wenleix told me that it's used for testing that the constructor doesn't throw. Either add a comment that says that or add some assertion to check that the constructed object has the correct state.

Reply:
Add a simple assertion after it. Thank you !!